event_session/wizards/wizard_event_session.py

```diff
@@ -0,0 +1,211 @@
+        help="Create sessions on Fridays",
+    )
+    saturdays = fields.Boolean(
+        help="Create sessions on Saturdays",
+    )
+    sundays = fields.Boolean(
+        help="Create sessions on Sundays",
+    )
+    delete_existing_sessions = fields.Boolean(
+        help="Check in order to delete every previous session for this event"
+    )
+    session_hour_ids = fields.One2many(
+        comodel_name='wizard.event.session.hours',
+        inverse_name='wizard_event_session_id',
+        string='Hours',
+    )
+
+    @api.multi
+    @api.constrains('session_hour_ids')
+    def _avoid_overlapping_hours(self):
+        for hour_a in self.session_hour_ids:
+            for hour_b in self.session_hour_ids:
+                if hour_a != hour_b:
+                    if hour_a.start_time == hour_b.start_time:
+                        raise ValidationError(
+                            _("There are overlapping hours!")
+                        )
+                    elif hour_b.start_time < \
+                            hour_a.start_time < hour_b.end_time:
+                        raise ValidationError(
+                            _("There are overlapping hours!")
+                        )
+
+    @api.multi
+    def weekdays(self):
+        return (self.mondays,
+                self.tuesdays,
+                self.wednesdays,
+                self.thursdays,
+                self.fridays,
+                self.saturdays,
+                self.sundays)
+
+    @api.multi
+    def datetime_fields(self):
+        """Fields converted to Python's Datetime-based objects."""
+        result = {
+            "event_start": fields.Datetime.from_string(self.event_date_begin),
+            "event_end": fields.Datetime.from_string(self.event_date_end),
+            "day_delta": timedelta(days=1),
+        }
+        return result
+
+    @api.multi
+    def existing_sessions(self, date):
+        """Return existing sessions that match some criteria."""
+        return self.env["event.session"].search(
+            (("event_id", "=", self.event_id.id),
+             ("date", "=", date),
+             ("start_time", "=", self.start_time)),
+        )
+
+    @api.multi
+    def create_session(self, **values):
+        """Create a new session record with the provided values."""
+        data = {
+            "name": "{} {} - {}".format(
+                strftime('%A %d/%m/%y',
+                         strptime(values["date"], "%Y-%m-%d %H:%M:%S")),
+                "%02d:%02d" % divmod(values["start_time"]*60, 60),
+                "%02d:%02d" % divmod(values["end_time"]*60, 60),
+                ),
+            "event_id": self.event_id.id,
+            "start_time": values["start_time"],
+            "end_time": values["end_time"],
+            "seats_min": self.event_id.seats_min,
+            "seats_max": self.event_id.seats_max,
+            "seats_availability": self.event_id.seats_availability,
+        }
+        data.update(values)
```

Comment:
If you do a method as "_prepare_session_values()" that return a dictionary other modules will can extend this method